src/Microsoft.Azure.Relay/HybridConnectionListener.cs

```diff
@@ -902,6 +902,7 @@ class DataConnection
             readonly HybridConnectionListener listener;
             readonly TokenProvider tokenProvider;
             readonly Uri rendezvousAddress;
+            readonly string subProtocol = "";
```

Comment:
Let's remove the ` = ""` from this declaration and wrap the call to `AddSubProtocol` with an `if (!string.IsNullOrEmpty(this.subProtocol))`.  Then I think this PR should be good to go.